packages/format-library/src/link/inline.js

```diff
@@ -96,6 +98,13 @@ class InlineLinkUI extends Component {
 			if ( opensInNewWindow !== state.opensInNewWindow ) {
 				return { opensInNewWindow };
 			}
+
+			if ( typeof rel === 'string' ) {
+				const noFollow = rel.split( ' ' ).includes( 'nofollow' );
+				if ( noFollow !== state.noFollow ) {
+					return { noFollow };
```

Comment:
What if `opensInNewWindow !== state.opensInNewWindow`? Then `noFollow` won't be derived?